### ISSUE TITLE
allow owners to leave projects

### ISSUE THREAD

**Issue by pabs3** [pypa/warehouse#4729]
**What's the problem this feature will solve?**

I'm one of the owners of the fonttools project and I would like to leave the project as it has been taken over by other folks and I am no longer involved. When I tried to do this the UI indicated it was not possible and trying it caused an error message.

**Describe the solution you'd like**

Allow people to leave projects if there are 1 or more other project owners.

**Comment by di** [comment]
If we were to implement this feature, we would need to change the way role assignment currently works, from just allowing owners to assign other users roles on a given project, to allowing owners to _request_ that other users accept roles for a given project.

Otherwise it would allow malicious users to create a (potentially malicious) project, assign it to some unsuspecting user, and then remove their ownership role, making it seem like the unsuspecting user was the original owner.

**Comment by pabs3** [comment]
I think it is important to require consent in response to invitations
to join a project, even if you do not allow owners to leave projects.
Being able to add someone to a project without their consent seems very
problematic and provides a way to spam people's profile pages.

-- 
bye,
pabs

https://bonedaddy.net/pabs3/

**Comment by di** [comment]
Blocked on #5790.

**Comment by bertjwregeer** [comment]
I just filed a duplicate: https://github.com/pypa/warehouse/issues/6365

**Comment by aaugustin** [comment]
I just hit this limitation while attempting to remove myself from the django-debug-toolbar project, in order to keep track with the reality of my involvement.

This used to be blocked on #5790 which is now resolved. If this is reasonably easy (like "just remove in two places the condition that prevents users from removing themselves and the corresponding tests") I could take a stab at it. Guidance appreciated obviously :-)

**Comment by di** [comment]
I agree this is no longer blocked. I think functionally this would be relatively easy, just removing a few conditionals:

https://github.com/pypa/warehouse/blob/54ebfc04a20291158839212855bdf8fa1ce33ba0/warehouse/templates/manage/roles.html#L108-L110

https://github.com/pypa/warehouse/blob/54ebfc04a20291158839212855bdf8fa1ce33ba0/warehouse/manage/views.py#L1817-L1819

...and adding a check that the user being removed is not a sole owner.

I think one important thing we should make sure to address is that it should be very clear when a user is removing themselves, perhaps with an additional warning.

Along these lines, we should also allow users to change their role from Owner to Maintainer or vice versa, if there's another owner.
